### PR TITLE
Add macOS build & tests to CircleCI.

### LINKED ISSUE #1238: Add macOS build & tests to CI.
Add macOS build & tests to CI.

### PR DESCRIPTION
**Pull request by PiotrSikora**
Fixes #1238.

### CODE REVIEW

**Comment by PiotrSikora** [comment]
Looks that we need switch to [macOS Plan](https://circleci.com/gh/organizations/istio/settings#osx-pricing) if we want test with macOS on CircleCI.

cc @louiscryan

File: .circleci/config.yml [hunk 1]
```diff
@@ -37,12 +36,31 @@ jobs:
...
+    environment:
+      - BAZEL_TEST_ARGS: "--test_env=ENVOY_IP_TEST_VERSIONS=v4only --test_output=all"
+    steps:
+      - run: sudo ntpdate -vu time.apple.com
+      - checkout
+      - restore_cache:
+          keys:
+            - bazel-cache-{{ checksum "WORKSPACE" }}-macos
+      - run: rm ~/.gitconfig
+      - run: bazel build //src/...
```
Comment (lizan) [inline .circleci/config.yml]: I think `make test` is enough?

Comment (PiotrSikora) [reply .circleci/config.yml]: Done.

**Comment by lizan** [comment]
ping, do we get macOS plan on CircleCI?

**Comment by PiotrSikora** [comment]
@sebastienvas what happened to CircleCI presubmits?

**Comment by sebastienvas** [comment]
@costinm I don't manage circle ci.

**Comment by costinm** [comment]
Sorry for the delay - we can get Mac plan, I think Louis' credit card is
needed.

Managing CircleCI is shared by all developers - the presubmit requirements
are
set by the admins in each project. Not sure who's admin for proxy - and how
this
is decided, but in practical terms proxy changes are following the standard
istio
suites and requirements when merging the SHA.

So proxy acts like a developent branch, with the large Istio pre-submit
checks
done when merging into istio main.  I don't think that's bad - or we should
be running
the full istio suites on proxy repo.

**Comment by louiscryan** [comment]
Enabled MacOS plan on Circle

**Comment by rshriram** [comment]
There is a bug in teh circleci config file.. dependencies.pre has to be converted into a plain step, that installs bazel and friends.. "Dependencies.pre" is a circleci 1.0 only config, while we are using circleci 2.0

**Pull request merged**